src/world/map.c

```diff
@@ -4503,77 +4503,91 @@ void map_extend_boundary_surface()
 }
 
 /**
- * Clears all elements properly from a certain tile.
- *  rct2: 0x0068AE2A
+ * Clears the provided element properly from a certain tile, and updates
+ * the pointer (when needed) passed to this function to point to the next element.
  */
-static void clear_elements_at(int x, int y)
+static void clear_element_at(int x, int y, rct_map_element **elementPtr)
 {
-	for (;;) {
-		for (int i = 0; i < 2; i++) {
-			rct2_peep_spawn *peepSpawn = &gPeepSpawns[i];
-			if (floor2(peepSpawn->x, 32) == x && floor2(peepSpawn->y, 32) == y) {
-				peepSpawn->x = UINT16_MAX;
-			}
-		}
-
-		rct_map_element *mapElement = map_get_first_element_at(x >> 5, y >> 5);
-	next_element:
-		switch (map_element_get_type(mapElement)) {
-		case MAP_ELEMENT_TYPE_SURFACE:
-			mapElement->base_height = 2;
-			mapElement->clearance_height = 2;
-			mapElement->properties.surface.slope = 0;
-			mapElement->properties.surface.terrain = 0;
-			mapElement->properties.surface.grass_length = 1;
-			mapElement->properties.surface.ownership = 0;
-			if (!map_element_is_last_for_tile(mapElement++))
-				goto next_element;
-
-			return;
-		case MAP_ELEMENT_TYPE_ENTRANCE:
-			viewport_interaction_remove_park_entrance(mapElement, x, y);
-			break;
-		case MAP_ELEMENT_TYPE_FENCE:
-			gGameCommandErrorTitle = STR_CANT_REMOVE_THIS;
-			game_do_command(
+	rct_map_element *element = *elementPtr;
+	switch (map_element_get_type(element)) {
+	case MAP_ELEMENT_TYPE_SURFACE:
+		element->base_height = 2;
+		element->clearance_height = 2;
+		element->properties.surface.slope = 0;
+		element->properties.surface.terrain = 0;
+		element->properties.surface.grass_length = 1;
+		element->properties.surface.ownership = 0;
+		// Because this element is not completely removed, the pointer must be updated muanually
+		// The rest of the elements are removed from the array, so the pointer doesn't need to be updated.
+		(*elementPtr)++;
+		break;
+	case MAP_ELEMENT_TYPE_ENTRANCE:
+		viewport_interaction_remove_park_entrance(element, x, y);
```

Comment:
You mean renaming `viewport_interaction_remove_park_entrance` to something like `remove_park_entrance`?
I don't know what this function does and, because of that, if that name is justified.